# HG changeset patch
# User Benjamin Lindner <lindnerben@gmx.net>
# Date 1233787710 18000
#      Wed Feb 04 17:48:30 2009 -0500
# Node ID 43c6012bd4c2046948113c3c56c515c7d24d9172
# Parent  0d263850d582c55d1f5f5d3200b249972648cbab
configure.in: separate cases for cygwin and mingw

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2009-02-04  Benjamin Lindner  <lindnerben@gmx.net>
+
+	* configure.in: Use separate cases for *-*-mingw* and *-*-cygwin*
+	to set variables related to creating shared libraries.
+
 2009-02-03  Jaroslav Hajek  <highegg@gmail.com>
 
 	* aclocal.m4 (AC_CHECK_QHULL_OK): New macro, based on suggestion by
 	Petr Mikulik <mikulik@physics.muni.cz>.
 	* configure.in: Call it.
 
 2009-02-01  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -1238,31 +1238,39 @@ case "$canonical_host_type" in
     SHLEXT=dylib 
     SHLLIB='$(SHLEXT)'
     SHLEXT_VER='$(version).$(SHLEXT)'
     SHLLIB_VER='$(version).$(SHLLIB)'
     NO_OCT_FILE_STRIP=true
     SONAME_FLAGS='-install_name $(octlibdir)/$@'
     library_path_var=DYLD_LIBRARY_PATH	
   ;;
-  *-*-mingw* | *-*-cygwin*)
-    case "$canonical_host_type" in
-      *-*-cygwin*)
-        LIBPRE=cyg
-	SHLLIBPRE=cyg
-        SHLBINPRE=lib
-      ;;
-    esac
+  *-*-cygwin*)
+    LIBPRE=cyg
+    SHLLIBPRE=cyg
+    SHLBINPRE=lib
     SHLEXT=dll
     SHLLIB=dll.a
     SHLBIN=a    
     DL_LDFLAGS="-shared -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--enable-runtime-pseudo-reloc"
     SH_LDFLAGS="-shared -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--enable-auto-image-base"
     SONAME_FLAGS='-Wl,--out-implib=$(patsubst $(SHLLIBPRE)%,$(SHLBINPRE)%,$@).a'
   ;;
+  *-*-mingw*)
+    CPICFLAG=
+    CXXPICFLAG=
+    FPICFLAG=
+    SHLEXT=dll
+    SHLLIB=dll.a
+    SHLBIN=dll
+    DL_LDFLAGS="-shared -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--enable-runtime-pseudo-reloc"
+    SH_LDFLAGS="-shared -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--enable-auto-image-base"
+    SONAME_FLAGS='-Wl,--out-implib=$@.a'
+    library_path_var=PATH
+  ;;
 
   *-*-msdosmsvc)
     DL_LDFLAGS="-shared"
     CPICFLAG=
     CXXPICFLAG=
     FPICFLAG=
     SHLEXT=dll
     SHLLIB=lib
